### ISSUE TITLE
Classifier src_key_padding_mask uses float tensor — mask does not prevent non-BGC genes from influencing classification as described in the paper

### ISSUE THREAD

**Issue by karman011** [HUST-NingKang-Lab/BGC-Prophet#7]
Hi, thank you for this great work!

While studying the classifier training code, I noticed that the gene-level tags (TDlabels) are passed as a float tensor to src_key_padding_mask in TransformerEncoder. According to PyTorch's documentation and implementation, this does not mask out non-BGC genes as described in the paper.

Paper description
"Transformer encoders allow key padding masks to be input to masks given specific timesteps; therefore, this study used gene tags as masks to prevent non-BGC genes from influencing the classification of BGC."

What happens in the code
In classifyTrainer.py:

```
sentence, labels, distribution = data[0], data[1], data[2]
# distribution is TDlabels: float tensor with 1.0 = BGC gene, 0.0 = non-BGC gene
output = self.model(src=sentence, src_key_padding_mask=distribution)
```
In classifier.py:

`memory = self.encoder(src, mask=None, src_key_padding_mask=src_key_padding_mask)`

TDlabels is returned from the dataset as torch.float32 with values 1.0 (BGC) and 0.0 (non-BGC).

**Comment by laiqilong** [comment]
Excellent catch! You are absolutely right that using a float tensor caused `src_key_padding_mask` to act as a soft bias rather than the intended hard mask.

However, since BGC-Prophet's primary focus is on BGC mining (predict), the practical impact of this on classification with few categories is minimal. The classification module is actually somewhat over-parameterized for its task. The ESM2 embeddings of just a few core enzymes provide such a strong signal that the model has the capacity to naturally learn to down-weight the noise from non-BGC genes during training, even without explicit hard masking.
